data/datasource.go

```diff
@@ -444,6 +465,53 @@ func readBoltDB(source *Source, args ...string) ([]byte, error) {
 	return data, nil
 }
 
+func readAWSSMP(source *Source, args ...string) ([]byte, error) {
+	log.Printf("In readAWSSMP with \nSOURCE=%#v\nURL=%#v\nARGS=%#v", source, source.URL, args)
+	if source.ASMPG == nil {
+		source.ASMPG = ssm.New(gaws.SDKSession())
+		source.Type = json_array_mimetype
+	}
+
+	p := source.URL.Path
+	recursive := false
+	if len(args) >= 1 {
+		// TODO: Note in docs that path.Join semantics are used (where are docs?)
+		// TODO: Can I add query param onto original URL & re-parse?
+		p = path.Join(p, args[0])
+		// TODO: Rather dirty magic word as 2nd arg to turn on recursion...
+		if len(args) >= 2 && "recursive" == args[1] {
+			recursive = true
+		}
+	}
+	input := &ssm.GetParametersByPathInput{
+		Path: aws.String(p),
+		Recursive: aws.Bool(recursive),
+		WithDecryption: aws.Bool(true),
+	}
+	log.Printf("INPUT=%#v", input)
+
+	results := make([]*ssm.Parameter, 0, 50)
+	// var results []*ssm.Parameter
+	err := source.ASMPG.GetParametersByPathPages(input,
```

Comment:
I know this is still a WIP and you're still trying different options, but I think for an initial cut I'd really like to see `GetParameter` be used instead.

I think this function shouldn't look too much different from the `readVault` function above (ln383)